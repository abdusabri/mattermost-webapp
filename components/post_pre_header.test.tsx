// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import {shallowWithIntl} from 'tests/helpers/intl-test-helper';
import PostPreHeader from 'components/post_pre_header';
import FlagIconFilled from 'components/widgets/icons/flag_icon_filled';

describe('components/PostPreHeader', () => {
    test('should not render anything if the post is neither flagged nor pinned', () => {
        const props = {
            isFlagged: false,
            isPinned: false,
        };

        const wrapper = shallowWithIntl(
            <PostPreHeader {...props}/>
        );

        expect(wrapper.find('div')).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });

    test('should not render anything if both skipFlagged and skipPinned are true', () => {
        const props = {
            isFlagged: true,
            isPinned: true,
            skipFlagged: true,
            skipPinned: true
        };

        const wrapper = shallowWithIntl(
            <PostPreHeader {...props}/>
        );

        expect(wrapper.find('div')).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });

    test('should properly handle flagged posts (and not pinned)', () => {
        const props = {
            isFlagged: true,
            isPinned: false,
        };

        const wrapper = shallowWithIntl(
            <PostPreHeader {...props}/>
        );

        expect(wrapper.find('span.icon-pin')).toHaveLength(0);

        expect(wrapper.find(FlagIconFilled)).toHaveLength(1);
        expect(wrapper.find(FormattedMessage)).toHaveLength(1);
        expect(wrapper.find(FormattedMessage).prop('defaultMessage')).toEqual('Flagged');
        expect(wrapper).toMatchSnapshot();

        // case of skipFlagged is true
        wrapper.setProps({...props, skipFlagged: true});

        expect(wrapper.find(FlagIconFilled)).toHaveLength(0);
        expect(wrapper.find(FormattedMessage)).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });

    test('should properly handle pinned posts (and not flagged)', () => {
        const props = {
            isFlagged: false,
            isPinned: true,
        };

        const wrapper = shallowWithIntl(
            <PostPreHeader {...props}/>
        );

        expect(wrapper.find(FlagIconFilled)).toHaveLength(0);

        expect(wrapper.find('span.icon-pin')).toHaveLength(1);
        expect(wrapper.find(FormattedMessage)).toHaveLength(1);
        expect(wrapper.find(FormattedMessage).prop('defaultMessage')).toEqual('Pinned');
        expect(wrapper).toMatchSnapshot();

        // case of skipPinned is true
        wrapper.setProps({...props, skipPinned: true});

        expect(wrapper.find('span.icon-pin')).toHaveLength(0);
        expect(wrapper.find(FormattedMessage)).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });

    describe('should properly handle posts that are flagged and pinned', () => {
        test('both skipFlagged and skipPinned are not true', () => {
            const props = {
                isFlagged: true,
                isPinned: true,
            };

            const wrapper = shallowWithIntl(
                <PostPreHeader {...props}/>
            );

            expect(wrapper.find(FlagIconFilled)).toHaveLength(1);
            expect(wrapper.find('span.icon-pin')).toHaveLength(1);
            expect(wrapper.find(FormattedMessage)).toHaveLength(1);
            expect(wrapper.find(FormattedMessage).prop('defaultMessage')).toEqual('Pinned and Flagged');
            expect(wrapper).toMatchSnapshot();
        });

        test('skipFlagged is true', () => {
            const props = {
                isFlagged: true,
                isPinned: true,
                skipFlagged: true,
            };

            const wrapper = shallowWithIntl(
                <PostPreHeader {...props}/>
            );

            expect(wrapper.find(FlagIconFilled)).toHaveLength(0);

            expect(wrapper.find('span.icon-pin')).toHaveLength(1);
            expect(wrapper.find(FormattedMessage)).toHaveLength(1);
            expect(wrapper.find(FormattedMessage).prop('defaultMessage')).toEqual('Pinned');
            expect(wrapper).toMatchSnapshot();
        });

        test('skipPinned is true', () => {
            const props = {
                isFlagged: true,
                isPinned: true,
                skipPinned: true,
            };

            const wrapper = shallowWithIntl(
                <PostPreHeader {...props}/>
            );

            expect(wrapper.find('span.icon-pin')).toHaveLength(0);

            expect(wrapper.find(FlagIconFilled)).toHaveLength(1);
            expect(wrapper.find(FormattedMessage)).toHaveLength(1);
            expect(wrapper.find(FormattedMessage).prop('defaultMessage')).toEqual('Flagged');
            expect(wrapper).toMatchSnapshot();
        });
    });
});
