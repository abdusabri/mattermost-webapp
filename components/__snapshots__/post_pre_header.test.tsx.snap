// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostPreHeader should not render anything if both skipFlagged and skipPinned are true 1`] = `<Fragment />`;

exports[`components/PostPreHeader should not render anything if the post is neither flagged nor pinned 1`] = `<Fragment />`;

exports[`components/PostPreHeader should properly handle flagged posts (and not pinned) 1`] = `
<div
  className="post-pre-header"
>
  <div
    className="post-pre-header__icons-container"
  >
    <FlagIconFilled
      className="icon icon--post-pre-header"
    />
  </div>
  <div
    className="post-pre-header__text-container"
  >
    <FormattedMessage
      defaultMessage="Saved"
      id="post_pre_header.flagged"
      values={Object {}}
    />
  </div>
</div>
`;

exports[`components/PostPreHeader should properly handle flagged posts (and not pinned) 2`] = `<Fragment />`;

exports[`components/PostPreHeader should properly handle pinned posts (and not flagged) 1`] = `
<div
  className="post-pre-header"
>
  <div
    className="post-pre-header__icons-container"
  >
    <span
      className="icon-pin icon icon--post-pre-header"
    />
  </div>
  <div
    className="post-pre-header__text-container"
  >
    <FormattedMessage
      defaultMessage="Pinned"
      id="post_pre_header.pinned"
      values={Object {}}
    />
  </div>
</div>
`;

exports[`components/PostPreHeader should properly handle pinned posts (and not flagged) 2`] = `<Fragment />`;

exports[`components/PostPreHeader should properly handle posts that are flagged and pinned both skipFlagged and skipPinned are not true 1`] = `
<div
  className="post-pre-header"
>
  <div
    className="post-pre-header__icons-container"
  >
    <span
      className="icon-pin icon icon--post-pre-header"
    />
    <FlagIconFilled
      className="icon icon--post-pre-header"
    />
  </div>
  <div
    className="post-pre-header__text-container"
  >
    <FormattedMessage
      defaultMessage="Pinned and Saved"
      id="post_pre_header.pinnedAndFlagged"
      values={Object {}}
    />
  </div>
</div>
`;

exports[`components/PostPreHeader should properly handle posts that are flagged and pinned skipFlagged is true 1`] = `
<div
  className="post-pre-header"
>
  <div
    className="post-pre-header__icons-container"
  >
    <span
      className="icon-pin icon icon--post-pre-header"
    />
  </div>
  <div
    className="post-pre-header__text-container"
  >
    <FormattedMessage
      defaultMessage="Pinned"
      id="post_pre_header.pinned"
      values={Object {}}
    />
  </div>
</div>
`;

exports[`components/PostPreHeader should properly handle posts that are flagged and pinned skipPinned is true 1`] = `
<div
  className="post-pre-header"
>
  <div
    className="post-pre-header__icons-container"
  >
    <FlagIconFilled
      className="icon icon--post-pre-header"
    />
  </div>
  <div
    className="post-pre-header__text-container"
  >
    <FormattedMessage
      defaultMessage="Saved"
      id="post_pre_header.flagged"
      values={Object {}}
    />
  </div>
</div>
`;
